crates/sui-types/src/deny_list.rs

```diff
@@ -56,3 +60,64 @@ pub struct PerTypeDenyList {
     // Table<String, VecSet<address>>
     pub denied_addresses: Table,
 }
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct CoinDenyCap {
+    pub id: UID,
+}
+
+impl DenyList {
+    pub fn check_coin_deny_list(
+        address: SuiAddress,
+        coin_types: BTreeSet<String>,
+        object_store: &dyn ObjectStore,
+    ) -> UserInputResult {
+        Self::check_deny_list(DENY_LIST_COIN_TYPE_INDEX, address, coin_types, object_store)
+    }
+
+    fn check_deny_list(
```

Comment:
can you write some comments describing the design of the deny list? or add a pointer to them if they already exist. It is a bit difficult to understand the motivation for the code here without a big picture.

Reply:
Add comments